pkg/solana/chainwriter/buffer_payload.go

```diff
@@ -119,7 +122,12 @@ func buildBufferExecutionReportIx(bufferID []byte, reportLen uint32, chunkPayloa
 	if ixErr != nil {
 		return nil, fmt.Errorf("failed to build buffer instruction: %w", ixErr)
 	}
-	return ix, nil
+	data, dataErr := ix.Data()
+	if dataErr != nil {
+		return nil, fmt.Errorf("failed to encode instruction data: %w", ixErr)
+	}
+	solanaIx := solana.NewInstruction(programID, ix.Accounts(), data)
```

Comment:
Qq, given that the buffer ix is always for the offramp program, is there a reason to allow overriding the invoked program? Or is it maybe just a convenience for some tests? Anyway, there's no problem with doing it, I was just curious about the rationale behind

Reply:
Ya this was because of tests. Doing `ccip_offramp.SetProgramID(offrampPubKey)` caused errors in tests when it was called multiple times. Then doing something like `ccip_offramp.ProgramID = offrampPubKey` caused some race issues in tests so I just went with setting it manually like this